### PR TITLE
addRepo improvements

### PR DESCRIPTION
**Pull request by dhoehna**
## Summary of the pull request
These improvements are in #2286.  That PR is frozen until design looks at it.  A minimum of 2 weeks.  Instead of waiting I moved the improvement code changes into this PR.
## Detailed description of the pull request / Additional comments
I moved more code out of the View's code-behind and into `AddRepoViewModel`.

The code in `AddRepoViewModel` is a bit messier and it still has a reference back to `AddRepoDialog`  However, this is a necessary step to get as much code out of `AddRepoDialog.xaml.cs` as possible.

## Validation steps performed
Cloned repos and added a Dev Drive.

Signed into github after clicking an account.

## PR checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

File: tools/SetupFlow/DevHome.SetupFlow/ViewModels/AddRepoViewModel.cs [hunk 1]
```diff
@@ -495,49 +602,77 @@ public void SetChangedEvents(TypedEventHandler<IDeveloperIdProvider, IDeveloperI
...
         IsAccountToggleButtonChecked = false;
         CurrentPage = PageKind.AddViaUrl;
         PrimaryButtonText = _stringResource.GetLocalized(StringResourceKey.RepoEverythingElsePrimaryButtonText);
         ShouldShowLoginUi = false;
+
+        ToggleCloneButton();
     }
 
-    public void ChangeToAccountPage()
+    public async Task ChangeToAccountPage()
```
Comment (krschau) [inline tools/SetupFlow/DevHome.SetupFlow/ViewModels/AddRepoViewModel.cs:620]:
```suggestion
    public async Task ChangeToAccountPageAsync()
```

File: tools/SetupFlow/DevHome.SetupFlow/ViewModels/AddRepoViewModel.cs [hunk 2]
```diff
@@ -495,49 +602,77 @@ public void SetChangedEvents(TypedEventHandler<IDeveloperIdProvider, IDeveloperI
...
+        IsUrlAccountButtonChecked = false;
+        IsAccountToggleButtonChecked = true;
+        CurrentPage = PageKind.AddViaAccount;
+        PrimaryButtonText = _stringResource.GetLocalized(StringResourceKey.RepoAccountPagePrimaryButtonText);
+        ShouldShowLoginUi = false;
+        ToggleCloneButton();
     }
 
-    public void ChangeToRepoPage()
+    public async Task ChangeToRepoPage()
```
Comment (krschau) [inline tools/SetupFlow/DevHome.SetupFlow/ViewModels/AddRepoViewModel.cs:661]:
```suggestion
    public async Task ChangeToRepoPageAsync()
```

**Review by krschau — APPROVED** [review]

File: tools/SetupFlow/DevHome.SetupFlow/ViewModels/AddRepoViewModel.cs (hunk 1 above)
Comment (dhoehna) [reply tools/SetupFlow/DevHome.SetupFlow/ViewModels/AddRepoViewModel.cs:620]: Changed.

File: tools/SetupFlow/DevHome.SetupFlow/ViewModels/AddRepoViewModel.cs (hunk 2 above)
Comment (dhoehna) [reply tools/SetupFlow/DevHome.SetupFlow/ViewModels/AddRepoViewModel.cs:661]: Changed.

**Review by vineeththomasalex — APPROVED** [review]

**Pull request merged**